sim/aviron/src/lib/io.zig

```diff
@@ -60,72 +61,40 @@ pub const Flash = struct {
 pub const RAM = struct {
     pub const Address = u24;
 
-    ctx: ?*anyopaque,
-    vtable: *const VTable,
-
-    /// Size of the RAM memory space in bytes.
-    size: usize,
-
-    pub fn read(mem: RAM, addr: Address) u8 {
-        std.debug.assert(addr < mem.size);
-        return mem.vtable.readFn(mem.ctx, addr);
-    }
-
-    pub fn write(mem: RAM, addr: Address, value: u8) void {
-        std.debug.assert(addr < mem.size);
-        return mem.vtable.writeFn(mem.ctx, addr, value);
-    }
-
-    pub const VTable = struct {
-        readFn: *const fn (ctx: ?*anyopaque, addr: Address) u8,
-        writeFn: *const fn (ctx: ?*anyopaque, addr: Address, value: u8) void,
-    };
-
-    pub const empty = RAM{
-        .ctx = null,
-        .size = 0,
-        .vtable = &VTable{ .readFn = empty_read, .writeFn = empty_write },
-    };
-
-    fn empty_read(ctx: ?*anyopaque, addr: u16) u8 {
-        _ = addr;
-        _ = ctx;
-        return 0;
-    }
-
-    fn empty_write(ctx: ?*anyopaque, addr: u16, value: u8) void {
-        _ = value;
-        _ = addr;
-        _ = ctx;
-    }
-
     pub fn Static(comptime size: comptime_int) type {
```

Comment:
Refactor `RAM.Static` into `FixedSizedMemory` and `RAM.Dynamic` into `FlexibleMemory` and remove the now unnecessary `RAM` type

Reply:
I kept Dynamic out of this PR, but I will change it to Flexible in a followup, and I will remove RAM for FixedSizedMemory